openquake/hazardlib/site.py

```diff
@@ -75,13 +80,14 @@ def __str__(self):
         >>> loc = openquake.hazardlib.geo.point.Point(1, 2, 3)
         >>> str(Site(loc, 760.0, True, 100.0, 5.0))
         '<Location=<Latitude=2.000000, Longitude=1.000000, Depth=3.0000>, \
-Vs30=760.0000, Vs30Measured=True, Depth1.0km=100.0000, Depth2.5km=5.0000>'
+Vs30=760.0000, Vs30Measured=True, Depth1.0km=100.0000, Depth2.5km=5.0000, \
+Backarc=False>'
         """
         return (
             "<Location=%s, Vs30=%.4f, Vs30Measured=%r, Depth1.0km=%.4f, "
-            "Depth2.5km=%.4f>") % (
+            "Depth2.5km=%.4f, Backarc=%r>") % (
             self.location, self.vs30, self.vs30measured, self.z1pt0,
-            self.z2pt5)
+            self.z2pt5, self.backarc)
```

Comment:
The backarc can be parameter used also in other GMPEs not just as a boolean parameter but also as a parameter defining a distance. For example, the Si and Midorikawa (1999) uses both the distances from the subduction trench and the volcanic arc (which are in principle also needed in this case to decide if you are in the backarc or forearc). I would therefore suggest to define this parameter as a float. In order to define positive distances I would suggest to assume that the volcanic trench axis follows the same direction (based on the right hand rule) used for the corresponding subduction interface. According to this (if I'm not mistaken) the distances to the volcanic will be positive on the left (i.e. down-dip side) and negative on other one (get_rx_distance https://github.com/gem/oq-hazardlib/blob/master/openquake/hazardlib/geo/surface/base.py#L88 can be used for this).